### PR TITLE
FUSETOOLS-2829 - provide information for other templates

### PR DESCRIPTION
**Pull request by apupier**
# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [x] Did you use the Jira Issue number in the commit comments?
- [x] Did you set meaningful commit comments on each commit?
- [x] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [x] Did the CI job report a successful build?
- [x] Is the non-happy path working, too?

## Maintainability

- [x] Are all Sonar reported issues fixed or can they be ignored?
- [x] Is there no duplicated code?
- [x] Are method-/class-/variable-names meaningful?

## Tests

- [x] Are there unit-tests?
- [x] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test? UI tests included in the PR

## Legal

- [x] Have you used the correct file header copyright comment?
- [x] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by apupier** [comment]
image
image

What do you think of this UI?
The big downside of it for now is that the text is not github links are not copiable. is it worthy to provide a first iteration like that? or it will bring too much frustration?

**Comment by apupier** [comment]
@apodhrad Can you have a check to the Red Deer code please? I found no "MessageDialog" out of the box so I created one. Relying on position of widget seems dangerous to me but I try to do similar thing than or TitleAreaDialog.
What do you think to split tests requiring a Fuse server and which are not?

**Comment by apupier** [comment]
@lhein @bfitzpat 

> What do you think of this UI?
> The big downside of it for now is that the text with github links is not copiable. Is it worthy to provide a first iteration like that? or it will bring too much frustration?

**Comment by apodhrad** [comment]
Hi @apupier, isn't it possible to display links instead of simple text?

File: uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/dialog/MessageDialog.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+	@Override
+	public Openable getDefaultOpenAction() {
+		return null;
+	}
+
+	public String getMessage() {
+		checkShell();
+		// Label message is the second Widget
+		return new DefaultLabel(((Label)handler.getChildren(getShell().getSWTWidget())[1])).getText();
```
Comment (apodhrad) [inline uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/dialog/MessageDialog.java]: I recomment to use a reference composite when looking for a widget + using index in the constructor, e.g
```
return new DefaultLabel(this, 1).getText();
```

File: uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/dialog/MessageDialog.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		super(shellTitle);
+	}
+
+	@Override
+	public Openable getDefaultOpenAction() {
+		return null;
+	}
+
+	public String getMessage() {
+		checkShell();
```
Comment (apodhrad) [inline uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/dialog/MessageDialog.java]: this is not necessary but you can use it, of course

File: uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/dialog/MessageDialog.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+package org.jboss.tools.fuse.reddeer.dialog;
+
+import org.eclipse.jface.window.Window;
+import org.eclipse.reddeer.core.handler.CompositeHandler;
+import org.eclipse.reddeer.jface.window.AbstractWindow;
+import org.eclipse.reddeer.jface.window.Openable;
+import org.eclipse.reddeer.swt.impl.label.DefaultLabel;
+import org.eclipse.swt.widgets.Label;
+
+public class MessageDialog extends AbstractWindow {
```
Comment (apodhrad) [inline uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/dialog/MessageDialog.java]: I suggest to use DefaultShell instad of AbstractWindow. Then, you don't need to specify the eclipse class and open action.

**Comment by apupier** [comment]
> isn't it possible to display links instead of simple text?

it is but requires to use a dedicated Wizard. it is also my concern, see https://github.com/jbosstools/jbosstools-fuse/pull/1280#issuecomment-367938107
I'm fine not to merge before implementing that (just don't know when it will be)

**Comment by apupier** [comment]
> @lhein @bfitzpat
> 
> What do you think of this UI?
> The big downside of it for now is that the text with github links is not copiable. Is it worthy to provide a first iteration like that? or it will bring too much frustration?

@lhein @bfitzpat up

**Comment by apupier** [comment]
put as "ready for review" given time constraint and that i get no functional feedback as asked.
Functional feedback is still welcome but now need also global review of the PR
@lhein @bfitzpat Please review

**Comment by apupier** [comment]
I created https://issues.jboss.org/browse/FUSETOOLS-2858 for next iteration

**Comment by lhein** [comment]
@apupier I am not a friend of that message dialog thing. Thats a model dialog in a wizard dialog. Also you can't click nor copy those links so the user is not really helped in terms of usability. 
How about leaving that label "Where can I find more examples?" in the wizard but instead:

Option 1: open a help page which provides clickable links -> will require help from @MelissaFlinn 
Option 2: do not open a dialog but make a text area visible at the bottom of the wizard which contains all the links and users are able to copy them

If you don't like any of those suggestions I would opt for not merging it until we can do it properly. I personally like Option 1 as it feels right to open a help page.

**Comment by apupier** [comment]
> Thats a modal dialog in a wizard dialog.

I don't think that's a problem for a first iteration and it was validated during UX review.

> Option 1: open a help page which provides clickable links -> will require help from @MelissaFlinn

I'm not sure what you mean by "open a help page", do you mean open this kind of right panel?
image


> Option 2: do not open a dialog but make a text area visible at the bottom of the wizard which contains all the links and users are able to copy them

it will introduce scrollbars.

Allowing to copy the link or even to open a browser page on click is possible at the bottom of the wizard or in a modal dialog too (just need to use our custom dialog and not a MessageDialog)

**Comment by lhein** [comment]
Yes, I think that approach from your screenshot looks suitable.

**Comment by apupier** [comment]
> Yes, I think that approach from your screenshot looks suitable.

this option means far more work. I don't think that it is suitable for a first iteration.

**Comment by apupier** [comment]
updated to provide a Selectable and copiable list, I really think that it is providing values to user even if a lot more can be done to go further.

image

**Comment by lhein** [comment]
+1 on green build. for first iteration that will do. as next step I would prefer the help solution.

**Comment by apupier** [comment]
wouhou test in success on devtools CI:
image

just have to fix the sonar reported issues

**Comment by apupier** [comment]
> as next step I would prefer the help solution.
I added link to your comment in follow-up issue: https://issues.jboss.org/browse/FUSETOOLS-2858

**Comment by apupier** [comment]
merge as green build and +1 in comment from @lhein

**Pull request merged**